app/Http/Controllers/DealController.php

```
@@ -2194,11 +2207,25 @@ protected function changeBulkStatus($request, ?array $editableIds = null)
             return;
         }
 
+        // Mass update bypasses DealObserver, so capture pre-change deals up front
+        // and notify their agents/watchers explicitly after the update (in-app
+        // only — mail stays suppressed by the bulk-action container flag).
+        $deals = Deal::whereIn('id', $editableIds)->get();
+
         if ($stage->slug === 'win' || $stage->slug === 'lost') {
```

PHPStan (2.2.7)

[error] Cannot access property $slug on array<App\Models\PipelineStage>|Illuminate\Database\Eloquent\Collection|Illuminate\Database\Eloquent\Model.


(property.nonObject)